http/peer-id-auth.md

```diff
@@ -52,55 +52,56 @@ examples.
 
 ## Base64 Encoding
 
-The base64 encoding follows Base 64 Encoding with URL and Filename Safe Alphabet
-from [RFC 4648](https://datatracker.ietf.org/doc/html/rfc4648#section-5). The
-reason this is not a multibase is to aid clients or servers who can not or
-prefer not to import a multibase dependency.
+The base64 encoding follows Base 64 Encoding with URL and Filename Safe
+Alphabet from [RFC
+4648](https://datatracker.ietf.org/doc/html/rfc4648#section-5). Padding MAY be
+omitted. The reason this is not a multibase is to aid clients or servers who
+can not or prefer not to import a multibase dependency.
 
-## Mutual Client and Server Peer ID
+## Mutual Client and Server Peer ID Authentication
 
 The following protocol allows both the client and server to authenticate each
-other's peer id by having them each sign a challenge issued by the other. The
+other's Peer ID by having them each sign a challenge issued by the other. The
 protocol operates as follows:
 
-1. The client makes a POST request to the authentication URI.
+1. The client makes an HTTP request to an authenticated resource.
 2. The server responds with status code 401 (Unauthorized) and set the header:
    ```
-   WWW-Authenticate: libp2p-PeerID challenge-client=<base64-encoded-challenge>, opaque=<base64-encoded-opaque-value>
+   WWW-Authenticate: libp2p-PeerID challenge-client="<challenge-string>", opaque="<base64-encoded-opaque-value>"
    ```
    The opaque parameter is opaque to client. The client MUST return the opaque
    parameter back to the server. The server MAY use the opaque parameter to
    encode state.
-3. The client makes another POST request to the authentication URI and sets the header:
+3. The client makes another HTTP request to the same authenticated resource and sets the header:
    ```
-   Authorization: libp2p-PeerID peer-id=<string-representation-of-client-peer-id>, opaque=<opaque-from-server>, challenge-server=<base64-encoded-challenge-server>, sig=<base64-signature-bytes>
+   Authorization: libp2p-PeerID peer-id="<string-representation-of-client-peer-id>", opaque="<opaque-from-server>", challenge-server="<challenge-string>"[,encoded-public-key="<base64-encoded-public-key-bytes>" ], sig="<base64-signature-bytes>"
    ```
 
+   The `encoded-public-key` param is optional and represents the client's public key. This is only needed when the client's public key is not included in the PeerID (e.g. not using the "identity" multihash).
+
    The `sig` param represents a signature over the parameters:
     - `hostname`
-    - `challenge-client` in its base64 encoded form.
+    - `challenge-client`
```

Comment:
If possible, I'd sign _everything_ the client sent us (basically, each step of the handshake should witness the previous step).

Reply:
This is what the client signs and the server has sent (challenge-client is the challenge for the client). The only thing here not signed that was sent by the server is `opaque` field. Why do you think that needs to be signed?

On the other side, the only thing the server doesn't sign is the returned `opaque` field as well. (and the string form of the peer id, but the server MUST verify that is equivalent/derived to/from the public key bytes we do sign)

Reply:
Would you prefer if the wording were changed to say something like "all parameters must be covered by the signature except for the `opaque` parameter, and the peer-id parameter (as it's covered by the public key bytes)"?